src/app/clusters/power-source-server/power-source-server.cpp

```diff
@@ -55,20 +105,111 @@ CHIP_ERROR PowerSourceAttrAccess::Read(const ConcreteReadAttributePath & aPath,
         // TODO: Needs implementation.
         err = aEncoder.EncodeEmptyList();
         break;
-    case EndpointList::Id:
-        // TODO: Needs implementation and a way to allow dynamic endpoints to register endpoints
-        err = aEncoder.EncodeEmptyList();
+    case EndpointList::Id: {
+        PowerSourceServer & server    = PowerSourceServer::Instance();
+        const Span<EndpointId> * span = server.GetEndpointList(aPath.mEndpointId);
+        if (span == nullptr)
+        {
+            err = aEncoder.EncodeEmptyList();
+        }
+        else
+        {
+            DataModel::List<EndpointId> list(*span);
+            err = aEncoder.Encode(list);
```

Comment:
This will not allow the list to be chunked.  You want to EncodeList() with the callback function that encodes item-at-a-time to allow chunking....  That's assuming you want to allow chunking in the middle of this list.  If we think this list will always fit in a packet, not chunking inside it just means it moves to the next packet, which is probably fine.

Reply:
I will change it to use EncodeList, but why do we have an encode function that takes a list argument if it is going to do the wrong thing?

Reply:
changed.

Reply:
It's not clear that "don't chunk" is always the wrong thing.  Sometimes it's the right thing, depending.